Tensile/Common.py

```diff
@@ -1617,14 +1618,22 @@ def pushWorkingPath( foldername ):
   return ensurePath( globalParameters["WorkingPath"] )
 def popWorkingPath():
   # Warning: this is not thread-safe, modifies the global WorkingPath!
-  globalParameters["WorkingPath"] = \
+  if len(workingDirectoryStack) == 0:
+    globalParameters["WorkingPath"] = \
       os.path.split(globalParameters["WorkingPath"])[0]
+  else:
+    globalParameters["WorkingPath"] = workingDirectoryStack.pop()
 def ensurePath( path ):
   try:
     os.makedirs(path)
   except OSError:
     pass
   return path
+def setWorkingPath( fullPathName ):
```

Comment:
I thought this is how `pushWorkingPath` actually worked.  Reviewing the calls to `pushWorkingPath`, I think this implementation could replace the old one, especially if we set the new directory to `os.path.join(cwd, newPath)` because if newPath is an absolute path it will just return `newPath`.